src/details/ArborX_DetailsDendrogram.hpp

```diff
@@ -89,33 +88,29 @@ template <typename ExecutionSpace, typename MemorySpace>
 void dendrogramUnionFind(ExecutionSpace const &exec_space,
                          Kokkos::View<WeightedEdge const *, MemorySpace> edges,
                          Kokkos::View<int *, MemorySpace> &parents,
-                         Kokkos::View<float *, MemorySpace> &parent_heights)
+                         Kokkos::View<float *, MemorySpace> &weights)
 {
   Kokkos::Profiling::pushRegion("ArborX::Dendrogram::dendrogram_union_find");
 
   auto const num_edges = edges.size();
   auto const num_vertices = num_edges + 1;
 
-  KokkosExt::reallocWithoutInitializing(exec_space, parent_heights, num_edges);
-
-  Kokkos::Profiling::pushRegion("ArborX::Dendrogram::edge_sort");
-  Kokkos::parallel_for(
-      "ArborX::Dendrogram::copy_weights",
-      Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, num_edges),
-      KOKKOS_LAMBDA(int const e) { parent_heights(e) = edges(e).weight; });
-  auto permute = sortObjects(exec_space, parent_heights);
-  Kokkos::View<UnweightedEdge *, MemorySpace> sorted_unweighted_edges(
+  KokkosExt::reallocWithoutInitializing(exec_space, weights, num_edges);
+  Kokkos::View<UnweightedEdge *, MemorySpace> unweighted_edges(
       Kokkos::view_alloc(
           exec_space, Kokkos::WithoutInitializing,
           "ArborX::Dendrogram::dendrogram_union_find::unweighted_edges"),
       num_edges);
   Kokkos::parallel_for(
-      "ArborX::Dendrogram::dendrogram_union_find::permute_and_fill",
+      "ArborX::Dendrogram::copy_weights_and_edges",
```

Comment:
Didn't mean to drop the `dendrogram_union_find`
Haven't thought too much whether the profiling regions needed more adjustment

Reply:
Should add back `dendrogram_union_find` for all labels and parallel constructs.

Reply:
Amending the fix